newrelic-agent/src/main/java/com/newrelic/agent/tracing/W3CTraceParentParser.java

```diff
@@ -7,19 +7,25 @@
 
 package com.newrelic.agent.tracing;
 
+import com.newrelic.agent.Agent;
 import com.newrelic.agent.MetricNames;
 import com.newrelic.agent.service.ServiceFactory;
 
 import java.util.List;
+import java.util.logging.Level;
 
 public class W3CTraceParentParser {
 
     static W3CTraceParent parseHeaders(List<String> traceParentHeaders) {
         if (traceParentHeaders.size() != 1) {
             ServiceFactory.getStatsService().getMetricAggregator().incrementCounter(MetricNames.SUPPORTABILITY_TRACE_CONTEXT_INVALID_PARENT_HEADER_COUNT);
-            return null;
+            Agent.LOG.log(Level.WARNING, "Multiple traceparent headers found on inbound request.");
+            // Multiple values ok if all are equal
+            boolean allHeadersEqual = traceParentHeaders.stream().allMatch(h -> h.equals(traceParentHeaders.get(0)));
```

Comment:
You might want to consider refactoring this to avoid using streams. The stream API can cause significantly more overhead when being used on small data structures than more traditional looping approaches. Ideally you should avoid using streams unless you're doing complex processing on a large data set.

Reply:
Noted, I think the modified loop is readable enough that performance should be the foremost concern. Updated to remove the streams API, plus a small refactoring and test for the empty list edge case.